x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -181,6 +287,39 @@ boolean isForceMergeIndex() {
         return steps;
     }
 
+    /**
+     * Resolves the prefix to be used for the mounted index depending on the provided key
+     */
+    String getRestoredIndexPrefix(StepKey currentKey) {
```

Comment:
this could be static

Reply:
These make use of the `storageType` instance variable, while we could passed it in, I think that would make the code unnecessarily verbose where these methods are called. I think I'd prefer to keep it as non-static if that's okay with you.